src/realm/util/scope_exit.hpp

```diff
@@ -43,20 +40,24 @@ class ScopeExit {
     ScopeExit(ScopeExit&& se) noexcept(std::is_nothrow_move_constructible<H>::value)
         : m_handler(std::move(se.m_handler))
     {
-        se.m_handler = none;
+        se.m_handler = std::nullopt;
     }
 
-    ~ScopeExit() noexcept
+    ~ScopeExit()
```

Comment:
Isn't it necessary to have a noexcept destructor? There is a static assert about this.

Reply:
Noexcept is the default for destructors and explicitly marking them as noexcept is cosmetic.